sonic_platform_base/sonic_storage/storage_devices.py

```diff
@@ -0,0 +1,71 @@
+#
+# storage_devices.py
+#
+# Composition class for the instantiation of objects of various storage device classes,
+# i.e. SsdUtil, EmmcUtil
+#
+
+
+try:
+    import os
+    import sys
+    from sonic_py_common import logger
+except ImportError as e:
+    raise ImportError (str(e) + "- required module not found")
+
+class StorageDevices():
+    def __init__(self, log_identifier):
+        self.devices = {}
+        self.BASE_PATH = "/sys/block"
+        self.BLKDEV_BASE_PATH = "/dev"
+        self._get_storage_devices()
+        self._get_storage_device_object()
+
+    def _get_storage_devices(self):
+        """
+        Function to get a list of storage disks in the switch by populating the self.devices dictionary with storage disk identifiers found in the /sys/block directory.
+
+        Args: N/A
+
+        Returns: N/A
+
+        """
+        fdlist = os.listdir(self.BASE_PATH)
+        for fd in fdlist:
+            if 'boot' in fd or 'loop' in fd:
+                continue
+            else:
+                self.devices[fd] = None
+
+    def _get_storage_device_object(self):
```

Comment:
@assrinivasan will be nice to return the object

Reply:
Do you mean that I can move the `for key in self.devices:` loop outside the function so that we can return an object or each key? If so, point taken, will make this change

Reply:
Done in latest commit